src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
```

Comment:
The `extraced_value` variable is totally unnecessary.  Remove it and just direct return.